modules/apps/27-interchain-accounts/keeper/keeper.go

```diff
@@ -86,16 +87,32 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s-%s", host.ModuleName, types.ModuleName))
 }
 
-// GetPort returns the portID for the interchain accounts module. Used in ExportGenesis
-func (k Keeper) GetPort(ctx sdk.Context) string {
+// GetAllPorts returns all bound ports for the interchain accounts module. Used in ExportGenesis
+func (k Keeper) GetAllPorts(ctx sdk.Context) []string {
 	store := ctx.KVStore(k.storeKey)
-	return string(store.Get([]byte(types.PortKey)))
+	iterator := sdk.KVStorePrefixIterator(store, []byte(types.PortKeyPrefix))
+	defer iterator.Close()
+
+	var ports []string
+	for ; iterator.Valid(); iterator.Next() {
+		keySplit := strings.Split(string(iterator.Key()), "/")
+
+		ports = append(ports, keySplit[1])
+	}
+
+	return ports
+}
+
+// HasPort returns true if the provided portID is found in state, otherwise false
+func (k Keeper) HasPort(ctx sdk.Context, portID string) bool {
```

Comment:
@colin-axner I've added `HasPort` as you suggested in a previous PR. It's not being used internally but I guess the idea is to expose this to auth modules?